### PR TITLE
Add k8s spec aspect

### PR DESCRIPTION
**Pull request by ddgenome**
Create aspect based on entire Kubernetes spec found in the top-level
of a repository.

### CODE REVIEW

**Comment by ddgenome** [comment]
I'm guessing I should add some stuff to the index.ts too.

**Comment by cdupuis** [comment]
@ddgenome is the intention that you create a different fingerprint per k8s object? That way you can only set a target on specific object. I can see how that would be useful across clusters.

File: lib/aspect/k8s/spec.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+/**
+ * Construct a Kubernetes spec fingerprint.
+ *
+ * @param spec Object representation of Kubernetes spec.
+ * @return Fingerprint of spec
+ */
+export function createK8sSpecFingerprint(spec: K8sObject): FP<K8sObject> {
+    return {
+        type: K8sSpecsType,
+        name: `${k8sSpecsFingerprintName(spec)}`,
```
Comment (cdupuis) [inline lib/aspect/k8s/spec.ts]: Why the quotes around a function call?

File: lib/aspect/k8s/spec.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    const resourceName = k8sSpecsFingerprintDisplayName(k8sSpecsFingerprintName(diff.from.data));
+    const repo = `${diff.owner}/${diff.repo}/${diff.branch}`;
+    return {
+        title: "New Kubernetes Spec Policy",
+        description: `Policy version for Kubernetes spec ${bold(resourceName)} differs from that in ${bold(repo)}.`,
+    };
+};
+
+/**
+ * Aspect emitting 0 or more npm dependencies fingerprints.
```
Comment (cdupuis) [inline lib/aspect/k8s/spec.ts]: Copy paste error

File: lib/aspect/k8s/spec.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+ */
+export const NpmDeps: Aspect<FP<K8sObject>> = {
+    displayName: "Kubernetes specs",
+    name: K8sSpecsType,
+    extract: createK8sSpecsFingerprints,
+    apply: applyK8sSpecsFingerprint,
+    summary: diffK8sSpecsFingerprints,
+    toDisplayableFingerprint: fp => fp.data[1],
```
Comment (cdupuis) [inline lib/aspect/k8s/spec.ts]: `data` is the spec object so array-index access won’t work I think. Realistically this should be a value that can be displayed in a PR body etc.

**Comment by cdupuis** [comment]
> I'm guessing I should add some stuff to the index.ts too.

Yes, add the aspect to `jobAspects` in index.ts and to the `Aspects` const.

File: lib/aspect/k8s/spec.ts (hunk 3 above)
Comment (ddgenome) [reply lib/aspect/k8s/spec.ts]: I'm not really sure how to concisely represent the "value" of an entire spec here, but it seems like that is what it wants. That is, the NpmDeps example shows the version there. I've just made it the display name.

**Comment by ddgenome** [comment]
@cdupuis yes the intent is to create a fingerprint for each resource spec in a sync repo. I've tried to make that more clear in the TypeDoc.

**Pull request merged**